scripts/globals/spells/spell_enhancing.lua

```diff
@@ -0,0 +1,344 @@
+        finalPower = finalPower + caster:getMerit(xi.merit.STORMSURGE) + caster:getMod(xi.mod.STORMSURGE_EFFECT)
+    end
+
+    ----------
+    -- Finish
+    ----------
+    return finalPower
+end
+
+-- Enhancing Spell Duration function.
+xi.spells.spell_enhancing.calculateEnhancingDuration = function(caster, target, spell, spellId, spellGroup, spellEffect)
+    local spellLevel   = enhancingTable[spellId][3]
+    local duration     = enhancingTable[spellId][5]
+    local useComposure = enhancingTable[spellId][6]
+    local targetLevel  = target:getMainLvl()
+
+    -- Deodorize, Invisible and Sneak have a random factor to base duration.
+    if spellEffect == xi.effect.DEODORIZE or spellEffect == xi.effect.INVISIBLE or spellEffect == xi.effect.SNEAK then
+        duration = duration + 60 * math.random(0, 2)
+    end
+
+    --------------------
+    -- Embolden
+    --------------------
+    if target:hasStatusEffect(xi.effect.EMBOLDEN) and spellGroup == xi.magic.spellGroup.WHITE then
+        duration = duration / 2
+    end
+
+    --------------------
+    -- Gear mods
+    --------------------
+    duration = duration + duration * caster:getMod(xi.mod.ENH_MAGIC_DURATION) / 100
+
+    ------------------------------
+    -- Merits and Job Points. (Applicable to all enhancing spells. Prior to multipliers, according to bg-wiki.)
+    ------------------------------
+    if caster:getMainJob() == xi.job.RDM then
+        duration = duration + caster:getMerit(xi.merit.ENHANCING_MAGIC_DURATION) + caster:getJobPointLevel(xi.jp.ENHANCING_DURATION)
+    end
+
+    --------------------------------------------------
+    -- Spell specific modifiers for duration.
+    --------------------------------------------------
+    -- Regen
+    if spellEffect == xi.effect.REGEN then
+        duration = duration + caster:getMod(xi.mod.REGEN_DURATION)
+        duration = duration + caster:getJobPointLevel(xi.jp.REGEN_DURATION) * 3
+
+    -- Invisible
+    elseif spellEffect == xi.effect.INVISIBLE then
+        duration = duration + target:getMod(xi.mod.INVISIBLE_DURATION)
+
+    -- Sneak
+    elseif spellEffect == xi.effect.SNEAK then
+        duration = duration + target:getMod(xi.mod.SNEAK_DURATION)
+    end
+
+    --------------------
+    -- Status Effects
+    --------------------
+    -- Composure
+    if useComposure and caster:hasStatusEffect(xi.effect.COMPOSURE) and caster:getID() == target:getID() then
+        duration = duration * 3
+    end
+
+    -- Perpetuance (Doesnt affect spikes and other Black magic enhancements)
+    if caster:hasStatusEffect(xi.effect.PERPETUANCE) and spellGroup == xi.magic.spellGroup.WHITE then
+        duration  = duration * 2
+    end
+
+    ------------------------------
+    -- Level penalty to duration.
+    ------------------------------
+    if targetLevel < spellLevel then
+        duration = duration * targetLevel / spellLevel
+    end
+
+    ----------
```

Comment:
Bit of an unnecessary comment at the end of this function

Reply:
Removed